app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/viewmodel/MainViewModel.java

```diff
@@ -0,0 +1,53 @@
+package org.dieschnittstelle.mobile.android.skeleton.viewmodel;
+
+import android.util.Log;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import org.dieschnittstelle.mobile.android.skeleton.model.IUserDatabaseOperation;
+import org.dieschnittstelle.mobile.android.skeleton.model.User;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MainViewModel extends ViewModel {
+        private IUserDatabaseOperation userDBOperation;
+        private MutableLiveData<String> loginLiveData = new MutableLiveData<>("initial");
```

Comment:
siehe TaskDetailViewModel. Hier soll wie ProceesingState implementiert werden.

```suggestion
        public enum LoginState {INVALID_EMAIL, WRONG_PASSWORD, AUTHENTICATION_SUCCESS, AUTHENTICATION_FAIL}
        private MutableLiveData<LoginState> loginState = new MutableLiveData<>();
```